### PR TITLE
Add support for tags and tagging

### PR DESCRIPTION
**Pull request by lavaturtle**
This adds support for interacting with the [tags](https://actionnetwork.org/docs/v2/tags) and [taggings](https://actionnetwork.org/docs/v2/taggings) endpoints of the ActionNetwork API.

Also included:
- Fixes a memoization bug with the `client.petitions(123).signatures` syntax
- Removes some redundant code from the `Signatures` class

### CODE REVIEW

File: lib/action_network_rest/client.rb [hunk 1]
```diff
@@ -32,7 +32,23 @@ def people
     end
 
     def petitions(petition_id=nil)
```
Comment (woodhull) [inline lib/action_network_rest/client.rb:34]: Should we DRY this pattern?

File: spec/client_spec.rb [hunk 2]
```diff
@@ -20,11 +20,49 @@
...
     it 'should work with a petition_id' do
-      expect(subject.petitions(123)).to be_a ActionNetworkRest::Petitions
+      p = subject.petitions(123)
+      expect(p).to be_a ActionNetworkRest::Petitions
+      expect(p.petition_id).to eq 123
+    end
+
+    it 'should work with two calls in a row with different arguments' do
+      p = subject.petitions
+      expect(p).to be_a ActionNetworkRest::Petitions
```
Comment (woodhull) [inline spec/client_spec.rb:36]: Is this valid, with no ID?

**Review by woodhull — APPROVED** [review]
This is starting to be really quite nice.

File: spec/client_spec.rb (hunk 2 above)
Comment (lavaturtle) [reply spec/client_spec.rb:36]: Yep, this is used for `client.petitions.create({some: 'data'})` calls

File: lib/action_network_rest/client.rb (hunk 1 above)
Comment (lavaturtle) [reply lib/action_network_rest/client.rb:34]: Hmm, not sure -- there's definitely some common structure, but I'm not sure how doable it is to DRY without sacrificing readability.

Comment (woodhull) [reply lib/action_network_rest/client.rb:34]: 👍

**Pull request merged**